hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/KeyDeletingService.java

```diff
@@ -288,6 +370,12 @@ private void resetMetrics() {
     getMetrics().setKdsCurRunTimestamp(latestRunTimestamp);
   }
 
+  // Helper: estimate protobuf serialized size of a string field
+  private static int estimateStringEntrySize(String key) {
+    int len = key.getBytes(StandardCharsets.UTF_8).length;
+    return 1 /* tag size */ + 1 /* length variant */ + len; /* actual string bytes */
```

Comment:
I feel this is an overkill of a logic and we are basically implementing something protobuf internal is not a great idea. We could use this instead
https://chromium.googlesource.com/external/github.com/google/protobuf/+/HEAD/java/core/src/main/java/com/google/protobuf/CodedOutputStream.java#:~:text=code%20string%7D%20field.%20*/-,public%20static%20int%20computeStringSizeNoTag(final%20String%20value)%20%7B,%7D,-/**
computeStringSizeNoTag is supposed to be used when the string is repeated string i.e. a list of string.